crates/transpiler/src/passes/consolidate_blocks.rs

```diff
@@ -349,7 +352,42 @@ pub fn run_consolidate_blocks(
     Ok(())
 }
 
+/// Replaces each block of consecutive gates by a single unitary node.
+///
+/// This is the main function of the `ConsolidateBlocks` transpiler pass
+/// which replaces uninterrupted sequences of gates acting on the same qubits
+/// into a Unitary node, which will consecutively be resynthesized into a
+/// more optimal subcircuit.
+///
+/// # Arguments
+/// * `dag` - The circuit for which we will consolidate gates.
+/// * `decomposer` - The type of decomposer to be used, could be of types
+///   [TwoQubitBasisDecomposer] or [TwoQubitControlledUDecomposer].
+/// * `basis_gate_name` - The basis gate selected for the decomposition.
+/// * `force_consolidate` - Decides whether to force all consolidations or not.
+/// * `target` - The target representing the backend for which the pass is consolidating.
+/// * `basis_gates` - The basis gates from which the decomposer will choose.
+pub fn run_consolidate_blocks(
+    dag: &mut DAGCircuit,
+    decomposer: DecomposerType,
+    basis_gate_name: &str,
+    force_consolidate: bool,
+    target: Option<&Target>,
+    basis_gates: Option<HashSet<String>>,
```

Comment:
If this is the rust entrypoint for running consolidate blocks this should take a target unconditionally and basis_gates as a set isn't a thing:

```suggestion
    target: &Target,
```

Reply:
I will add this to the follow-on PR.

Reply:
Although, there are cases where you can run the Pass alone without a `Target` and it would just choose a `TwoQubitBasisDecomposer` with a `CXGate` for a base. Are these not going to be applicable from Rust?

Reply:
I think having an optional target is fine, there is a use case for this in the preset pass managers too (during the init stage). I had forgotten about that when I left the comment yesterday. But we don't need the basis gates set it should only be a target or none.

Reply:
This was removed from the follow-up PR, so we should be good to go.